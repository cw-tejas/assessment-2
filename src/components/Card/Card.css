/* .card {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    background-color: #f2f2f2;
    background-color: #c13939;
    border-radius: 5%;
}

.thumbnail {
    width: 300px;
    height: 200px;
} */


.card {
    width: 100%;
    border-radius: 10px;
    border: 1px solid #ccc;
}

.card .card-image {
    width: 100%;
    height: auto;
    object-fit: cover;
    border-radius: 10px 10px 0px 0px;
}

.card-content {
    padding: 16px;
    text-align: start;
}

.category {
    font-size: 12px;
    letter-spacing: 1px;
    color: #9CA3AF;
    margin-bottom: 8px;
}

.card-title {
    font-size: 15px;
    font-weight: 600;
    color: #111827;
    margin-bottom: 12px;
    margin-top: 0px;
}

.card-description {
    font-size: 14px;
    color: rgb(72, 72, 72);
    margin-bottom: 16px;
}

.card-price {
    font-size: 18px;
    /* color: #3076d2; */
    font-weight: 600;
    margin-bottom: 16px;
}

.card-make-offer {
    font-size: 16px;
    color: #3076d2;
    font-weight: 600;
    margin-bottom: 16px;
}

.card-footer {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
}

.learn-more {
    color: #6366F1;
    display: inline-flex;
    align-items: center;
    text-decoration: none;
    font-size: 14px;
    cursor: pointer;
    margin-bottom: 8px;
}

.learn-more:hover {
    text-decoration: underline;
}

.arrow-icon {
    width: 16px;
    height: 16px;
    margin-left: 8px;
}

.views, .comments {
    color: #9CA3AF;
    display: inline-flex;
    align-items: center;
    font-size: 14px;
    margin-right: 12px;
    padding-right: 12px;
    border-right: 2px solid #E5E7EB;
}

.comments {
    border-right: none;
}

.view-icon, .comment-icon {
    width: 16px;
    height: 16px;
    margin-right: 4px;
}

.btn {
    background-color: rgb(239, 42, 42);
    color: white;
    display: block;
    width: 100%;
    text-align: center;
    border-radius: 5px;
    padding: 10px;
    font-size: 15px;
}
