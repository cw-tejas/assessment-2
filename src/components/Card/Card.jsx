import React from "react";
import "./Card.css";

const Card = ({car}) => {
  return (
    <div className="card">
      <img className="card-image" src={car.imageUrl || "https://dummyimage.com/300x225/000/fff"} alt="blog" />
      <div className="card-content">
        <h2 className="card-title">{car.carName}</h2>
        <p className="card-description">{car.km} km | {car.fuel} | {car.cityName}</p>
        <p className="card-price">{car.price}</p>
        <p className="card-make-offer">Make Offer</p>
        <div className="card-footer">
          <a className="btn">Get Seller Details</a>
        </div>
      </div>
    </div>
  );
};

export default Card;
