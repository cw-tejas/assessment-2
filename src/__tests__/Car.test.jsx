import { render, screen } from "@testing-library/react";
import Card from "../components/Card/Card";

const carMock = {
  profileId: "1",
  carName: "Car A",
  km: "2000",
  fuel: "Petrol",
  cityName: "City A",
  price: "1000000",
  imageUrl: "car-a.jpg",
};

describe("Card Component", () => {
  it("renders car information", () => {
    render(<Card car={carMock} />);
    expect(screen.getByText(/Car A/i)).toBeInTheDocument();
    expect(screen.getByText(/2000 km/i)).toBeInTheDocument();
    expect(screen.getByText(/Petrol/i)).toBeInTheDocument();
    expect(screen.getByText(/1000000/i)).toBeInTheDocument();
  });
  
  it("renders default image when image URL is missing", () => {
    const carWithoutImage = { ...carMock, imageUrl: "" };
    render(<Card car={carWithoutImage} />);
    expect(screen.getByAltText(/blog/i).src).toContain("dummyimage.com");
  });

  // snapshot
  it("matches snapshot", () => {
    const { asFragment } = render(<Card car={carMock} />);
    expect(asFragment()).toMatchSnapshot();
  })
});
