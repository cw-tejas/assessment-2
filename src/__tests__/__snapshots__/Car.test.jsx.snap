// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="blog"
      class="card-image"
      src="car-a.jpg"
    />
    <div
      class="card-content"
    >
      <h2
        class="card-title"
      >
        Car A
      </h2>
      <p
        class="card-description"
      >
        2000 km | Petrol | City A
      </p>
      <p
        class="card-price"
      >
        1000000
      </p>
      <p
        class="card-make-offer"
      >
        Make Offer
      </p>
      <div
        class="card-footer"
      >
        <a
          class="btn"
        >
          Get Seller Details
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
